client/app/components/form/UserOperatorMultiStepForm.tsx

```diff
@@ -0,0 +1,91 @@
+"use client";
+
+import { RJSFSchema } from "@rjsf/utils";
+import { useState } from "react";
+import { useParams, useRouter, useSearchParams } from "next/navigation";
+import { userOperatorUiSchema } from "@/app/utils/jsonSchema/userOperator";
+import { actionHandler } from "@/app/utils/actions";
+import MultiStepFormBase from "@/app/components/form/MultiStepFormBase";
+import {
+  UserOperatorFormData,
+  UserFormData,
+} from "@/app/components/form/formDataTypes";
+
+interface UserOperatorFormProps {
+  readonly schema: RJSFSchema;
+  readonly formData: Partial<UserFormData>;
+}
+
+export default function UserOperatorMultiStepForm({
+  schema,
+  formData,
+}: UserOperatorFormProps) {
+  const { push } = useRouter();
+  const params = useParams();
+  const searchParams = useSearchParams();
+  const [error, setError] = useState(undefined);
+  const [formState, setFormState] = useState(formData);
+
+  const formSection = parseInt(params?.formSection as string) - 1;
+
+  const formSectionList = Object.keys(schema.properties as RJSFSchema);
+  const isFinalStep = formSection === formSectionList.length - 1;
+
+  const submitHandler = async (data: { formData?: UserOperatorFormData }) => {
+    const newFormData = {
+      ...formState,
+      ...data.formData,
+    } as UserOperatorFormData;
+
+    // to prevent resetting the form state when errors occur
+    setFormState(newFormData);
+
+    // add user operator id to form data if it exists (to be used in senior officer creation)
+    const userOperatorId = searchParams.get("user-operator-id");
+    if (userOperatorId) newFormData.user_operator_id = userOperatorId;
+
+    const apiUrl = `registration/user-operator/${
+      isFinalStep ? "contact" : "operator"
+    }`;
+
+    const response = await actionHandler(
+      apiUrl,
+      "POST",
+      `/dashboard/select-operator/user-operator/create/${params?.formSection}`,
+      {
+        body: JSON.stringify(newFormData),
+      },
+    );
+
+    if (response.error) {
+      setError(response.error);
+      return;
+    }
+
+    if (isFinalStep) {
+      push(
+        `/dashboard/select-operator/received/add-operator/${response.operator_id}`,
+      );
+      return;
+    }
+
+    push(
```

Comment:
FYI this is a different url pattern than the operations multistep urls. That form uses operations/0/1 (0 is a placeholder for the operation id and becomes operations/[id]/2 after the first step) and 1 and 2 are the step numbers. 

Maybe a tech debt card to standardize them both into the best of both worlds? I think `create` is more helpful/descriptive than 0, and [id]/[step] is easier to read than `user-operator/create/2?user-operator-id=4`